stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1574,21 +1580,61 @@ public class ChatClient internal constructor(
         private val tokenManager: TokenManager = TokenManagerImpl()
         private var customOkHttpClient: OkHttpClient? = null
 
+        /**
+         * Sets the log level to be used by the client.
+         *
+         * See [ChatLogLevel] for details about the available options.
+         *
+         * We strongly recommend using [ChatLogLevel.NOTHING] in production builds,
+         * which produces no logs.
+         *
+         * @param level The log level to use.
+         */
         public fun logLevel(level: ChatLogLevel): Builder {
             logLevel = level
             return this
         }
 
+        /**
+         * Sets a [ChatLoggerHandler] instance that will receive log events from the SDK.
+         *
+         * Use this to forward SDK events to your own logging solutions.
+         *
+         * See the FirebaseLogger class in the UI Components sample app for an example implementation.
+         *
+         * @param loggerHandler Your custom [ChatLoggerHandler] implementation.
+         */
         public fun loggerHandler(loggerHandler: ChatLoggerHandler): Builder {
             this.loggerHandler = loggerHandler
             return this
         }
 
+        /**
+         * Sets a custom [ChatNotificationHandler] that the SDK will use to handle everything
+         * around push notifications. Create your own subclass and override methods to customize
+         * notification appearance and behavior.
+         *
+         * See the [Push Notifications](https://staging.getstream.io/chat/docs/sdk/android/client/guides/push-notifications/)
+         * documentation for more information.
+         *
+         * @param notificationsHandler Your custom subclass of [ChatNotificationHandler].
+         */
         public fun notifications(notificationsHandler: ChatNotificationHandler): Builder {
             this.notificationsHandler = notificationsHandler
             return this
         }
 
+        /**
+         * Sets a custom file uploader implementation that will be used by the client
+         * to upload files and images.
+         *
+         * The default implementation uses Stream's own CDN to store these files,
+         * which has a 20 MB upload size limit.
```

Comment:
We need to review this ASAP, I'm almost sure clients will still not be able to send files via UI components as this limit is hardcoded there.

Reply:
I believe this is handled with the [`streamUiAttachmentMaxFileSizeMb`](https://github.com/GetStream/stream-chat-android/blob/6e6354373b6011dedc4b152f4aac04bd68fe05f5/stream-chat-android-ui-components/src/main/res/values/attrs_message_input_view.xml#L94) attribute of `MessageInputView`. That value gets passed into `MessageInputFieldView`, which seems to use it correctly.

Am I missing something else? Should we also handle this somewhere else in the code?

Reply:
Sorry, you are right. Just always get confused about this line in our adapter:
```
internal var attachmentMaxFileSize: Long = AttachmentConstants.MAX_UPLOAD_FILE_SIZE
```

Reply:
Ah yes, that's just by default, but it should be updated correctly when it's overridden.